### PR TITLE
Adds SearchGuildMemberRequest

### PR DESCRIPTION
**Pull request by Shadorc**
**Justification:** https://github.com/discord/discord-api-docs/pull/1577

### CODE REVIEW

**Review by quanticc — APPROVED** [review]

File: src/main/java/discord4j/discordjson/json/SearchGuildMembersRequest.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+@JsonDeserialize(as = ImmutableSearchGuildMembersRequest.class)
+public interface SearchGuildMembersRequest {
+
+    static ImmutableSearchGuildMembersRequest.Builder builder() {
+        return ImmutableSearchGuildMembersRequest.builder();
+    }
+
+    String query();
+
+    Possible<Optional<Integer>> limit();
```
Comment (darichey) [inline src/main/java/discord4j/discordjson/json/SearchGuildMembersRequest.java:21]: Not nullable

**Review by darichey — CHANGES_REQUESTED** [review]

File: src/main/java/discord4j/discordjson/json/SearchGuildMembersRequest.java (hunk 1 above)
Comment (Shadorc) [reply src/main/java/discord4j/discordjson/json/SearchGuildMembersRequest.java:21]: `All parameters to this endpoint except for `query` are optional`

Comment (Shadorc) [reply src/main/java/discord4j/discordjson/json/SearchGuildMembersRequest.java]: Fixed

**Comment by Shadorc** [comment]
This pull request may not be needed anymore: https://github.com/Discord4J/Discord4J/pull/888#issuecomment-817049173
Feel free to close it if you decide to use the `Map` approach

**Comment by quanticc** [comment]
Ah that's right, query parameters are being passed through map, and request body through data classes.

**Pull request closed without merging**